New logs

Pietervanhalem opened the pull request.

hoonhout reviewed (commented):
This PR lacks a bit of explanation. It seems that you just want to make the log messages more explanatory by adding object names to the messages. But in the mean time you remove functionality. Can you comment?

Pietervanhalem commented on openclsim/core.py:
```diff
@@ -1601,12 +1601,29 @@ def __init__(self, v: float = 1, *args, **kwargs):
+        )
+        
+        container_level = self.container.get_level() if hasattr(self, "container") else 0
         # Log the start event
-        self.log_sailing(log_state=LogState.START)
+        self.log_entry(
+            message,
+            self.env.now,
+            container_level,
+            self.geometry,
+            self.ActivityID,
+            LogState.START,
+        )
```
see general comment.

Pietervanhalem commented:
Nanco pointed out that in the current situation all the logs have a specified format. {ActivityType} {ActivityName} {status}. From all perspectives (Activity logs, site logs and vessel logs) this is consistent, except for shift amount and sailing. 

The logging of the sailing (only from the vessel perspective) is the most weird. It does not mention either ActivityType or ActivityName and it only mentions state (sailing empty / sailing full). However sailing full or empty is ambiguous since it can have mobilisation, transit, demobilisation. In the first openclsim we did not have this logging issue and just sailing empty / sailing full was enough. For sailing holds that this state was IMHO determined in a weird way. ```state = full if container.level != 0 else empty```. This is In many cases just not true. For dredging we almost always have rest load, and for offshore wind we have transits. Therefore I removed the ```self.log_sailing()``` function and replaced it with what we actually know. 

For shift amounth it was a bit less weird but still not consistent. Here I just changed it to the general format.

hoonhout reviewed (approved):
Ok, that makes sense. The log line for move activities contained an interpretation of the container level that was not really relevant for the event described, if not incorrect. Instead, the container level is simply added to each log line.

Approved as-is, but still 2 questions:
1) Why does the VO-COLOS implementation not inherit from this one? For example, why is the `get_container_level` implemented twice? And
2) Why did you remove the support for multi containers? Can't we keep it in by allowing the container level to be a dict? For example:

```python
    def get_container_level(self):
        if hasattr(self, "container"):
            if hasattr(self.container, "container_list"):
                return {id_: self.container.get_level(id_) for id_ in self.container.container_list}
            else:
                return self.container.get_level()
        else:
            return -1 
```

Or make it a `dict` in all cases, if that is easier. Anyway, you get the idea.

Pietervanhalem commented:
First I do not inherit Movable since the Movable inherits from Locatable. And openclsim.core.Locatable needs to be initialised with a non serialisable object (shaply point) and my Locatable needs to be initialised with lat, lon (json serialisable). Im sure there is some python black magic to fix this, but I just copied it.

I do inherit from the processor for BasicVessel and BasicVesselLoadingFunction. However not for the Interface vessels. For this interface I rewrote so much of the processor that it would only be confusing for me to do the with inheritance. 

I did not mean to break the multicontainer and I don't think I did break the multicontainer. It is just the logging, multicontainers work as good as they ever did.  For reference I pasted the old logging of the old Moveable below. Here they also just log ```self.container.get_level()```. I think it still logs the same information and IMHO I don't think we should pass dicts as values of the log.

I think this PR keeps the functionality the same, even for the multi containers. However, I think there is still some work to do to implement the multi containers (I am just very unfamiliar with it).  I think we should do this in a separate PR.

```python
def log_sailing(self, log_state):
        """ Log the start or stop of the sailing event """

        if isinstance(self, HasContainer):
            list_ = self.container.container_list
            status = (
                "filled"
                if len(list_) > 0
                and sum([self.container.get_level(id_) for id_ in list_]) > 0
                else "empty"
            )
            self.log_entry(
                "sailing {}".format(status),
                self.env.now,
                self.container.get_level(),
                self.geometry,
                self.ActivityID,
                log_state,
            )
        else:
            self.log_entry(
                # "sailing {}".format(event),
                "sailing",
                self.env.now,
                -1,
                self.geometry,
                self.ActivityID,
                log_state,
            )

```

hoonhout commented:
Ok, but then we might want to consider to simply use the model classes from OpenCLSim and the core classes from VOCOLOS. And leave the core classes in OpenCLSim untouched for now. Would that work? I'm struggling with how these packages interact and don't like code redundancy, as you know ;-)

hoonhout commented:
And for future reference: indeed you didn't break the multicontainer, and the old logging method indeed did not log multicontainers correctly either. Just be aware that `self.container.get_level()` on a multicontainer only returns the level of the `default` container. It will be zero if you define, for example, a `MP` and `TP` container.
